pkg/middleware/arm_client_opts.go

```diff
@@ -33,7 +33,10 @@ func DefaultArmOpts(userAgent string, logCollector ArmRequestMetricCollector, cu
 		runtime.NewRequestIDPolicy(),
 		&ArmRequestMetricPolicy{Collector: logCollector},
 	}
-	opts.PerCallPolicies = customPerCallPolicies
+	opts.PerCallPolicies = []policy.Policy{}
+	if customPerCallPolicies != nil {
+		opts.PerCallPolicies = append(opts.PerCallPolicies, customPerCallPolicies...)
+	}
```

Comment:
Do you think we can add a assertion test case for this DefaultARMOpts to validate that in the future we don't have a regression that breaks the customPercallpolicies assignment?

Reply:
We should start teesting all of these fields